Source/WebKit/webpushd/PushService.h

```diff
@@ -57,10 +57,15 @@ class PushService {
     PushServiceConnection& connection() { return m_connection; }
     WebCore::PushDatabase& database() { return m_database; }
 
+    Vector<String> enabledTopics() { return m_connection->enabledTopics(); }
```

Comment:
Ideally this function would be const and return a `const Vector<String>`

Reply:
This eventually bottoms out in `PushServiceConnection::enabledTopics()`, which has to return a `Vector<String>` rather than `const Vector<String>&`. This is because `ApplePushServiceConnection::enabledTopics()` (which is another subclass in this class hierarchy and the one that actually matters in the real implementation) has to return `Vector<String>` since it doesn't cache the topics locally in a member variable. Instead it always constructs a `Vector<String>` on the fly by converting it from the `NSArray<NSString *>` given to it from APS. So there is always at least one copy involved.

Given this, making the return type `Vector<String>` and counting on move semantics to move the copy up the stack seems okay to me.